web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerStatus/DocumentConnectMyComputerStatus.tsx

```diff
@@ -0,0 +1,203 @@
+              display: flex;
+            `}
+          >
+            <Laptop mr={2} />
+            {/** The node name can be changed, so it will be different from the "system hostname." */}
+            {agentNode?.hostname || hostname}
+          </Text>
+          <MenuIcon
+            buttonIconProps={{
+              css: css`
+                border-radius: ${props => props.theme.space[1]}px;
+                background: ${props => props.theme.colors.spotBackground[0]};
+              `,
+            }}
+            menuProps={{
+              anchorOrigin: {
+                vertical: 'bottom',
+                horizontal: 'right',
+              },
+              transformOrigin: {
+                vertical: 'top',
+                horizontal: 'right',
+              },
+            }}
+          >
+            <MenuItem onClick={() => alert('Not implemented')}>
+              Remove agent
+            </MenuItem>
+          </MenuIcon>
+        </Flex>
+
+        <Transition in={!!agentNode} timeout={1_800} mountOnEnter>
+          {state => (
+            <LabelsContainer gap={1} className={state}>
+              {renderLabels(agentNode.labelsList)}
+            </LabelsContainer>
+          )}
+        </Transition>
+        <Flex mt={3} mb={3} display="flex" alignItems="center">
+          {prettyAgentState.title}
+        </Flex>
+        {prettyAgentState.error && (
+          <Alert
+            css={`
+              white-space: pre-wrap;
+            `}
+          >
+            {prettyAgentState.error}
+          </Alert>
+        )}
+        <Text mb={4}>
+          Connecting your computer will allow any cluster user with the role{' '}
+          <strong>{roleName}</strong> to access it as an SSH resource with the
+          user <strong>{systemUsername}</strong>.
+        </Text>
+        {state.status === 'running' || state.status === 'stopping' ? (
+          <ButtonPrimary
+            block
+            disabled={state.status === 'stopping'}
+            onClick={kill}
+          >
+            Disconnect
+          </ButtonPrimary>
+        ) : (
+          <ButtonPrimary
+            block
+            disabled={state.status === 'starting'}
+            onClick={runWithPreparation}
+          >
+            Connect
+          </ButtonPrimary>
+        )}
+      </Box>
+    </Document>
+  );
+}
+
+function renderLabels(labelsList: tsh.Label[]): JSX.Element[] {
+  const labels = labelsList.map(makeLabelTag);
+  return labels.map(label => (
+    <Label key={label} kind="secondary">
+      {label}
+    </Label>
+  ));
+}
+
+function prettifyAgentState(agentState: AgentState): {
+  title: string;
+  error?: string;
+} {
+  switch (agentState.status) {
+    case 'starting':
+      return { title: '🔄 Starting' };
+    case 'stopping':
+      return { title: '🔄 Stopping' };
+    case 'not-started': {
+      return { title: '🔘 Agent not running' };
+    }
+    case 'running': {
+      return { title: '🟢 Agent running' };
```

Comment:
Those emojis were there only for the mockups, let's replace them with some proper icons.